### PR TITLE
Compiled plan support for property, parameter, literal, and addition projections.

### PR DESCRIPTION
**Pull request by pontusmelke**
Code generation support for:
- `MATCH n RETURN n.prop`
- `MATCH a-[r]->b RETURN r.prop`
- `RETURN 1`
- `MATCH a RETURN a.prop + 1`
- `RETURN [1, 2, 3] + [4, 5]`
- `RETURN [1, 2, 3] + 4`
- `RETURN "FOO" + 4`
- …

### CODE REVIEW

File: community/cypher/cypher-compiler-2.3/src/main/java/org/neo4j/cypher/internal/compiler/v2_3/birk/Javac.java [hunk 1]
```diff
@@ -90,12 +92,12 @@
     }
 
     public static InternalExecutionResult newInstance( Class<InternalExecutionResult> clazz, Statement statement,
-                                                       GraphDatabaseService db, InternalPlanDescription description)
+                                                       GraphDatabaseService db, ExecutionMode executionMode, InternalPlanDescription description, HashMap<String, Object> params)
```
Comment (lutovich) [inline community/cypher/cypher-compiler-2.3/src/main/java/org/neo4j/cypher/internal/compiler/v2_3/birk/Javac.java]: I think just `Map<String, Object>` should be better

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/il/ProjectionInstruction.scala [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  def projectedVariable = JavaSymbol(s"ro.nodeGetProperty( $nodeIdVar, $propKeyVar ).value( null )", OBJECT)
+}
+
+case class ProjectRelProperty(token: Option[Int], propName: String, relIdVar: String, namer: Namer) extends ProjectionInstruction {
+  private val propKeyVar = token.map(_.toString).getOrElse(namer.next())
+
+  def generateInit() =
+    if (token.isEmpty)
+      s"""if ( $propKeyVar == -1 )
+                            |{
```
Comment (lutovich) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/il/ProjectionInstruction.scala]: Code formatting got broken here

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/il/ProjectionInstruction.scala [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+case class ProjectLiteral(literal: AnyRef) extends ProjectionInstruction {
+
+  def generateInit() = ""
+
+  def projectedVariable = JavaSymbol(s"$literal", OBJECT)
+
+  def fields() = ""
+}
+
+case class ProjectNodeProperties(projections:Seq[ProjectionInstruction], parent:Instruction) extends Instruction {
```
Comment (lutovich) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/il/ProjectionInstruction.scala]: This class should probably be renamed to smth like `ProjectProperties` as in `CodeGenerator` it is used as aggregator for both node and relationship properties

**Comment by pontusmelke** [comment]
@systay moved the semantic table. Feel free to do any changes you see fit, I'll not have any time to do more fixes until I get back

**Pull request merged**